src/NexusMods.App.UI/Windows/WindowManager.cs

```diff
@@ -73,4 +80,49 @@ public void UnregisterWindow(IWorkspaceWindow window)
 
         _allWindowIdSource.Edit(list => list.Remove(window.WindowId));
     }
+
+    public void SaveWindowState(IWorkspaceWindow window)
+    {
+        try
+        {
+            var data = window.WorkspaceController.ToData();
+            _dataStore.Put(WindowData.Id, data);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Exception while saving window state");
+        }
+    }
+
+    public bool RestoreWindowState(IWorkspaceWindow window)
+    {
+        var isFirstWindow = _windows.Count == 1;
+        if (!isFirstWindow) return false;
```

Comment:
Does this mean that if you open a new MainWindow while you have an existing one open it will start from scratch but still save when closed?

Would there be a problem with always restoring here?

Reply:
When a main window closes, we save the data to the data store. Since the data is written to the same location every time, the windows will overwrite whatever existing data is there. This means that only the data of the last closed main window will be restored.

As far as restoring goes, we only want to restore data for the first main window. Any subsequently opened main windows are going to have a clean slate.

Reply:
That means that if you close one of the clean slates last you will loose all your setup.
Why shouldn't a new window also load the last saved state?